### PR TITLE
chore(dbapi2): deprecate trace_fetch_methods configuration

### PR DESCRIPTION
**Pull request by P403n1x87**
## Description

As part of the effort of deprecating the dbapi2 configuration, this change deprecates the trace_fetch_methods configuration item.

## Checklist
- [ ] Added to the correct milestone.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

**Review by brettlangdon — APPROVED** [review]

File: ddtrace/contrib/dbapi/__init__.py [hunk 1]
```diff
@@ -176,11 +177,20 @@ class TracedConnection(wrapt.ObjectProxy):
...
+            cfg = config.dbapi2
         # Set default cursor class if one was not provided
         if not cursor_cls:
             # Do not trace `fetch*` methods by default
             cursor_cls = TracedCursor
-            if config.dbapi2.trace_fetch_methods:
+            # Deprecation of config.dbapi2 requires we add a check
+            if cfg.trace_fetch_methods or config.dbapi2.trace_fetch_methods:
+                if config.dbapi2.trace_fetch_methods:
+                    debtcollector.deprecate(
```
Comment (majorgreys) [inline ddtrace/contrib/dbapi/__init__.py:189]: Would be great to include a test for this deprecation warning.

File: ddtrace/contrib/psycopg/patch.py [hunk 2]
```diff
@@ -69,7 +79,13 @@ def __init__(self, conn, pin=None, cursor_cls=None):
         if not cursor_cls:
             # Do not trace `fetch*` methods by default
             cursor_cls = Psycopg2TracedCursor
-            if config.dbapi2.trace_fetch_methods:
+            if config.psycopg.trace_fetch_methods or config.dbapi2.trace_fetch_methods:
```
Comment (P403n1x87) [inline ddtrace/contrib/psycopg/patch.py:82]: @majorgreys @brettlangdon what are the odds someone would set `config.psycopg.trace_fetch_methods` to something other than `True`? If this is explicitly set to `False`, and `config.dbapi2.trace_fetch_methods` is set to `True`, then we would have a bug.

Comment (brettlangdon) [reply ddtrace/contrib/psycopg/patch.py:82]: something other than `True` is probably fairly low, but "truthy" check here should be good enough.

> If this is explicitly set to False, and config.dbapi2.trace_fetch_methods is set to True, then we would have a bug

you mean, if `config.psycopg.trace_fetch_methods = False` and `config.dbapi2.trace_fetch_methods = True`?

Comment (majorgreys) [reply ddtrace/contrib/psycopg/patch.py:82]: Right, the default for these being `False` makes deciding here difficult. Someone explicitly disabling `config.psycopg.trace_fetch_methods` but still enabling via `config.psycopg.trace_fetch_methods` would find fetch methods traced still. Upon examining the logs they'd come across the deprecation warning and hopefully have already seen the release notes for using `config.psycopg.trace_fetch_methods`, in which case they'd have a clear resolution (not setting `config.psycopg.trace_fetch_methods`). The present behavior then doesn't seem to be a bug but an inconvenience that's coming from the deprecation process.

File: ddtrace/contrib/dbapi/__init__.py [hunk 3]
```diff
@@ -69,7 +70,7 @@ def _trace_method(self, method, name, resource, extra_tags, *args, **kwargs):
 
             # set analytics sample rate if enabled but only for non-FetchTracedCursor
             if not isinstance(self, FetchTracedCursor):
-                s.set_tag(ANALYTICS_SAMPLE_RATE_KEY, config.dbapi2.get_analytics_sample_rate())
+                s.set_tag(ANALYTICS_SAMPLE_RATE_KEY, cfg.get_analytics_sample_rate())
```
Comment (brettlangdon) [inline ddtrace/contrib/dbapi/__init__.py:73]:
```suggestion
                s.set_tag(ANALYTICS_SAMPLE_RATE_KEY, self._self_config.get_analytics_sample_rate())
```

?

File: releasenotes/notes/deprecate-dbapi-config-0fc5b1acfc4aded2.yaml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+---
+deprecations:
+  - |
+    Deprecate `ddtrace.config.dbapi2` as default for `TracedCursor` and `TracedConnection` as well as `DD_DBAPI2_TRACE_FETCH_METHODS`. Use `IntegrationConfig` and `DD_<INTEGRATION>_TRACE_FETCH_METHODS` specific to each dbapi-compliant library.
```
Comment (brettlangdon) [inline releasenotes/notes/deprecate-dbapi-config-0fc5b1acfc4aded2.yaml:4]: should we provide an example here, like:

```python
# Before
config.dbapi2.trace_fetch_methods = True

# After
config.psycopg2.trace_fetch_methods = True
```

?

File: ddtrace/contrib/dbapi/__init__.py (hunk 3 above)
Comment (P403n1x87) [reply ddtrace/contrib/dbapi/__init__.py]: Ah! I didn't push my latest change where I'd fixed this!

File: releasenotes/notes/deprecate-dbapi-config-0fc5b1acfc4aded2.yaml (hunk 4 above)
Comment (P403n1x87) [reply releasenotes/notes/deprecate-dbapi-config-0fc5b1acfc4aded2.yaml:4]: 👍 sounds good

**Review by Kyle-Verhoog — APPROVED** [review]
:+1: lgtm

**Review by Yun-Kim — APPROVED** [review]

**Pull request merged**